File: backend/src/main/java/com/godeltech/mastery/backend/repository/OperationRepository.java
package com.godeltech.mastery.backend.repository;

import com.godeltech.mastery.backend.domain.entity.Car;
import com.godeltech.mastery.backend.domain.entity.ServiceOperationRecord;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface OperationRepository extends JpaRepository<ServiceOperationRecord, Long> {

  Optional<List<ServiceOperationRecord>> getAllByCar(Car car);
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/repository/OperationRepository.kt
package com.godeltech.mastery.backend.repository

import com.godeltech.mastery.backend.domain.entity.Car
import com.godeltech.mastery.backend.domain.entity.ServiceOperationRecord
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface OperationRepository : JpaRepository<ServiceOperationRecord, Long> {

    fun getAllByCar(car: Car): Optional<List<ServiceOperationRecord>>
}